pkg/operator/reconciler.go

```diff
@@ -0,0 +1,402 @@
+package operator
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	errors "k8s.io/apimachinery/pkg/api/errors"
+	k8serros "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	tunedv1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/manifests"
+	ntomf "github.com/openshift/cluster-node-tuning-operator/pkg/manifests"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/metrics"
+	"github.com/openshift/cluster-node-tuning-operator/version"
+)
+
+const ClusterOperatorName = "node-tuning"
+
+// Reconciler reconciles a Tuned object and creates the tuned daemon set.
+type Reconciler struct {
+	Client    client.Client
+	Scheme    *runtime.Scheme
+	Namespace string
+}
+
+// SetupWithManager creates a new PerformanceProfile Controller and adds it to the Manager.
```

Comment:
If I understand correctly, this  SetupWithManager creates a new Tuned Controller and adds it to the manager. 
```suggestion
// SetupWithManager creates a new Tuned Controller and adds it to the Manager.
```

Reply:
The responsibilities of the controllers are commented above each `Reconcile` func.
This is the template controller runtime had in the past but we can change it if we wish.

Reply:
I think for this PR, we should at least correct the reference to `PerformanceProfile` in this comment.

Reply:
already done in the last commit https://github.com/openshift/cluster-node-tuning-operator/pull/302/commits/329389d756c0e6317ce4d13401fde7c25a98a914